### PR TITLE
[POC] add deprecation warning to remove/replace them later

### PR DESCRIPTION
**Pull request by ElectricMaxxx**
| Q             | A
| ------------- | ---
| Branch?       | master for bug fixes
| Bug fix?      | no
| New feature?  | no 
| BC breaks?    | yes
| Deprecations? | yes
| Tests pass?   | yes - "Tabität"
| Fixed tickets | #24835 
| License       | MIT

This will be my first proposal: I try to evaluate where to use the class parameters from. At the we have to deprecate from next version on and release 5.0 to really remove the usage. The given proposal means either use a get merhod or pass it through object manager config.

### CODE REVIEW

File: src/Symfony/Bridge/Doctrine/DependencyInjection/AbstractDoctrineExtension.php [hunk 1]
```diff
@@ -352,6 +414,12 @@ protected function loadCacheDriver($cacheName, $objectManagerName, array $cacheD
             case 'xcache':
             case 'wincache':
             case 'zenddata':
+                if ($container->hasParameter($this->getObjectManagerElementName(sprintf('cache.%s.class', $cacheDriver['type'])))) {
+                    @trigger_error(
+                        'Usage of class parameter for '.$cacheDriver['type'].' class is deprecated since version 4.2 and
+                            ill be removed in 5.0. Use configuration values instead.'
```
Comment (nicolas-grekas) [inline src/Symfony/Bridge/Doctrine/DependencyInjection/AbstractDoctrineExtension.php]: typo: will
please also put all call to trigger_error on a single line to have them consistent with the code base and easy grepping the code

Comment (ElectricMaxxx) [reply src/Symfony/Bridge/Doctrine/DependencyInjection/AbstractDoctrineExtension.php]: OK i'll fix that, any idea what would be the right usage now? I mean i will have to write it in the dreprecation notice.

Comment (nicolas-grekas) [reply src/Symfony/Bridge/Doctrine/DependencyInjection/AbstractDoctrineExtension.php]: Maybe some inspiration in #11881?

**Comment by ElectricMaxxx** [comment]
#De ahta sie seich schon weiter entwicklt. Vll sind sie noch erreichbar.
lg max

**Comment by nicolas-grekas** [comment]
Back on this. Before reviewing in depth, could you please rebase and revert *all* code style changes? They make reviewing more difficult and don't follow the CS we use actually.
Thanks :)

**Comment by ElectricMaxxx** [comment]
I will do tonight.

File: src/Symfony/Bridge/Doctrine/DependencyInjection/AbstractDoctrineExtension.php [hunk 2]
```diff
@@ -360,7 +438,7 @@ protected function loadCacheDriver($cacheName, $objectManagerName, array $cacheD
 
         $cacheDef->setPublic(false);
 
-        if (!isset($cacheDriver['namespace'])) {
+        if (!isset($cacheDriver['namesp ace'])) {
```
Comment (stof) [inline src/Symfony/Bridge/Doctrine/DependencyInjection/AbstractDoctrineExtension.php]: that's broken

File: src/Symfony/Bridge/Doctrine/DependencyInjection/AbstractDoctrineExtension.php [hunk 3]
```diff
@@ -352,7 +410,12 @@ protected function loadCacheDriver($cacheName, $objectManagerName, array $cacheD
             case 'xcache':
             case 'wincache':
             case 'zenddata':
-                $cacheDef = new Definition('%'.$this->getObjectManagerElementName(sprintf('cache.%s.class', $cacheDriver['type'])).'%');
+                if ($container->hasParameter($this->getObjectManagerElementName(sprintf('cache.%s.class', $cacheDriver['type'])))) {
+                    @trigger_error(
+                        'Usage of class parameter for '.$cacheDriver['type'].' class is deprecated since version 4.2 and will be removed in 5.0. Use configuration values instead.'
+                    );
```
Comment (chalasr) [inline src/Symfony/Bridge/Doctrine/DependencyInjection/AbstractDoctrineExtension.php]: `trigger_error` calls miss a `E_USER_DEPRECATED` as 2nd argument

**Comment by nicolas-grekas** [comment]
@ElectricMaxxx how should we proceed with this PR? Maybe it's not worth it?

File: src/Symfony/Bridge/Doctrine/DependencyInjection/AbstractDoctrineExtension.php [hunk 4]
```diff
@@ -28,7 +28,6 @@ abstract class AbstractDoctrineExtension extends Extension
      * Used inside metadata driver method to simplify aggregation of data.
      */
     protected $aliasMap = array();
-
```
Comment (stof) [inline src/Symfony/Bridge/Doctrine/DependencyInjection/AbstractDoctrineExtension.php]: this empty line should be kept

File: src/Symfony/Bridge/Doctrine/DependencyInjection/AbstractDoctrineExtension.php [hunk 5]
```diff
@@ -319,10 +348,26 @@ protected function loadCacheDriver($cacheName, $objectManagerName, array $cacheD
...
+                        E_USER_DEPRECATED
+                    );
+                }
+                if ($container->hasParameter($this->getObjectManagerElementName('cache.memcached_instance.class'))) {
+                    @trigger_error(
+                        'Usage of class parameter for memcached_instance class is deprecated since version 4.2 and will be removed in 5.0. Use configuration values instead.',
+                        E_USER_DEPRECATED
+                    );
+                }
+                $memcachedClass = !empty($cacheDriver['class']) ? $cacheDriver['class'] : '%' . $this->getObjectManagerElementName('cache.memcached.class') . '%';
```
Comment (stof) [inline src/Symfony/Bridge/Doctrine/DependencyInjection/AbstractDoctrineExtension.php]: We should not expect the class to come from the config data. Each bundle knows what the classes are. We don't want them to make it configurable in their semantic config (that would be even worse than the DI parameter).

**Comment by fabpot** [comment]
Closing as there is no more feedback here. Feel free to reopen.

**Pull request closed without merging**